### PR TITLE
improve sentinel, add tests

### PR DESCRIPTION
**Pull request by xiaozhou**
- fix threshold updates when add or remove guards
- change `updateGuardState` function to explicit `guard()` and `relax()` functions
- add tests

### CODE REVIEW

File: contracts/safeguard/sentinel/Guard.sol [hunk 1]
```diff
@@ -29,28 +29,31 @@ abstract contract Guard is Ownable {
...
+        require(guardStates[msg.sender] == GuardState.Guarded, "invalid caller");
+        guardStates[msg.sender] = GuardState.Relaxed;
+        numRelaxedGuards++;
         _updateRelaxed();
-        emit GuardUpdated(msg.sender, _state);
+        emit GuardUpdated(msg.sender, GuardState.Relaxed);
     }
 
-    function addGuards(address[] calldata _accounts, bool _incrementThreshold) external onlyOwner {
+    function addGuards(address[] calldata _accounts, uint256 _thresholdIncrement) external onlyOwner {
```
Comment (Dominator008) [inline contracts/safeguard/sentinel/Guard.sol:50]:
```solidity    
    function addGuards(address[] calldata _accounts, uint256 _newRelaxThreshold) external onlyOwner {
        for (uint256 i = 0; i < _accounts.length; i++) {
            _addGuard(_accounts[i]);
        }
        setRelaxThreshold(_newRelaxThreshold);
    }

    function removeGuards(address[] calldata _accounts, uint256 _newRelaxThreshold) external onlyOwner {
        for (uint256 i = 0; i < _accounts.length; i++) {
            _removeGuard(_accounts[i]);
        }
        setRelaxThreshold(_newRelaxThreshold);
    }
 ```
 Cleaner?

Comment (xiaozhou) [reply contracts/safeguard/sentinel/Guard.sol]: merged into a single cleaner `updateGuards` function to minimize owner operation overhead

File: contracts/safeguard/sentinel/Guard.sol [hunk 2]
```diff
@@ -29,30 +29,36 @@ abstract contract Guard is Ownable {
...
-    function addGuards(address[] calldata _accounts, bool _incrementThreshold) external onlyOwner {
-        for (uint256 i = 0; i < _accounts.length; i++) {
-            _addGuard(_accounts[i]);
+    function updateGuards(
+        address[] calldata _add,
+        address[] calldata _remove,
+        uint256 _newRelaxThreshold
+    ) external onlyOwner {
+        for (uint256 i = 0; i < _add.length; i++) {
+            _addGuard(_add[i]);
```
Comment (Dominator008) [inline contracts/safeguard/sentinel/Guard.sol:56]: nit: remove before add?

Comment (xiaozhou) [reply contracts/safeguard/sentinel/Guard.sol]: done

**Review by Dominator008 — APPROVED** [review]

**Pull request merged**